packages/ui/src/components/va-affix/VaAffix.vue

```diff
@@ -16,168 +16,143 @@
-      const target = this.getTargetElement()
+      const target = getTargetElement()
 
-      if (this.offsetTop === undefined) {
-        return
-      }
+      if (props.offsetTop === undefined) { return }
 
-      if (target !== window) {
-        const { top } = (target as HTMLElement).getBoundingClientRect()
-        return top + this.offsetTop
+      if (!(target instanceof Window)) {
+        const { top } = target.getBoundingClientRect()
+        console.log(top)
+        console.log(props.offsetTop)
+        return top + props.offsetTop
       }
 
-      return this.offsetTop
+      return props.offsetTop
     }
 
     const calculateBottom = () => {
-      const target = this.getTargetElement()
-      if (this.offsetBottom === undefined) {
-        return
-      }
+      const target = getTargetElement()
+
+      if (props.offsetBottom === undefined) { return }
 
-      if (target !== window) {
-        const { bottom } = (target as HTMLElement).getBoundingClientRect()
-        const { offsetHeight, clientHeight } = (target as HTMLElement)
+      if (!(target instanceof Window)) {
+        const { bottom } = target.getBoundingClientRect()
+        const { offsetHeight, clientHeight } = target
         const scrollBarHeight = offsetHeight - clientHeight
         const windowHeight = getWindowHeight()
-        return windowHeight - (bottom - this.offsetBottom) + scrollBarHeight
+        return windowHeight - (bottom - props.offsetBottom) + scrollBarHeight
       }
 
-      return this.offsetBottom
+      return props.offsetBottom
     }
 
     const convertToPixels = (calculate: () => number | undefined) => {
       const result = calculate()
-
-      if (result === undefined) {
-        return
-      }
-
-      return `${result}px`
+      return result === undefined ? undefined : `${result}px`
     }
 
-    return {
-      top: this.state.isTopAffixed ? convertToPixels(calculateTop) : null,
-      bottom: this.state.isBottomAffixed ? convertToPixels(calculateBottom) : null,
-      width: `${this.state.width}px`,
-    }
-  }
-
-  get isAffixed (): boolean {
-    return this.state.isTopAffixed || this.state.isBottomAffixed
-  }
-
-  handleThrottledEvent (eventName: string | null, event?: Event) {
-    const context: Context = {
-      ...this.$data,
-      ...this.$props,
-      element: this.$refs.element as any,
-      target: this.getTargetElement(),
-      setState: this.setState.bind(this),
-      getState: this.getState.bind(this),
-    }
-
-    if (!eventName || eventName === 'resize') {
-      handleThrottledEvent(eventName, context)
-    } else if (event && event.target) {
-      const target = this.getTargetElement()
+    const computedClass = computed(() => [{ 'va-affix--affixed': isAffixed }])
+    const computedStyle = computed(() => ({
+      top: state.value.isTopAffixed ? convertToPixels(calculateTop) : null,
+      bottom: state.value.isBottomAffixed ? convertToPixels(calculateBottom) : null,
```

Comment:
Better ts typing for CSSProperties type.

```suggestion
      top: state.value.isTopAffixed ? convertToPixels(calculateTop) : undefined,
      bottom: state.value.isBottomAffixed ? convertToPixels(calculateBottom) : undefined,
```